src/data/zmod/basic.lean

```diff
@@ -440,3 +440,29 @@ instance : discrete_field (zmodp p hp) :=
   ..zmodp.has_inv hp }
 
 end zmodp
+
+namespace int
+
+lemma add_even_of_add_square_even {x y : ℤ} (h : (x^2 + y^2) % 2 = 0) : (x + y) % 2 = 0 :=
```

Comment:
This should have a one-line proof. It should be split in `x % 2 = x^2 % 2` and `x^2 + y^2 % 2 = (x + y)^2 % 2`. Of course I am assuming that `% 2` takes values in `{0,1}`. But maybe it can also sometimes be `-1`?